addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/handler/HyperionHandler.java

```diff
@@ -0,0 +1,315 @@
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.hyperion.internal.connection.JsonTcpConnection;
+import org.openhab.binding.hyperion.internal.protocol.ColorCommand;
+import org.openhab.binding.hyperion.internal.protocol.CommandUnsuccessfulException;
+import org.openhab.binding.hyperion.internal.protocol.EffectCommand;
+import org.openhab.binding.hyperion.internal.protocol.HyperionCommand;
+import org.openhab.binding.hyperion.internal.protocol.ServerInfoCommand;
+import org.openhab.binding.hyperion.internal.protocol.v1.ActiveEffect;
+import org.openhab.binding.hyperion.internal.protocol.v1.ActiveLedColor;
+import org.openhab.binding.hyperion.internal.protocol.v1.ClearAllCommand;
+import org.openhab.binding.hyperion.internal.protocol.v1.ClearCommand;
+import org.openhab.binding.hyperion.internal.protocol.v1.Effect;
+import org.openhab.binding.hyperion.internal.protocol.v1.Transform;
+import org.openhab.binding.hyperion.internal.protocol.v1.TransformCommand;
+import org.openhab.binding.hyperion.internal.protocol.v1.V1Info;
+import org.openhab.binding.hyperion.internal.protocol.v1.V1Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link HyperionHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniel Walters - Initial contribution
+ */
+public class HyperionHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HyperionHandler.class);
+
+    private JsonTcpConnection connection;
+    private ScheduledFuture<?> refreshFuture;
+    private ScheduledFuture<?> connectFuture;
+    private Gson gson = new Gson();
+
+    private static final ServerInfoCommand SERVER_INFO_COMMAND = new ServerInfoCommand();
+
+    private String address;
+    private int port;
+    private int refreshInterval;
+    private int priority;
+
+    private Runnable refreshJob = new Runnable() {
+        @Override
+        public void run() {
+            try {
+                V1Response response = sendCommand(SERVER_INFO_COMMAND);
+                if (response.isSuccess()) {
+                    handleServerInfoResponse(response);
+                }
+            } catch (IOException e) {
+                logger.warn("Could not connect to server.");
```

Comment:
reduce to debug, see [E.4](https://docs.openhab.org/developers/development/guidelines.html#e-logging)